### PR TITLE
Potential Code Re-Factor and Bug Fixes

### PR DESCRIPTION
**Pull request by HarshCasper**
## Summary

This pull request removes the Dangerous Default Argument and the assert statements from the Non-test Files. It also allowed Dictionary iteration directly. 

## Changes

This Pull Request has these Commits: 

- Remove assert statement from non-test files
- Fix dangerous default argument
- Iterate dictionary directly

## Importance

I ran a DeepSource Analysis on my fork of this repository. You can see all the issues raised by DeepSource [here](https://deepsource.io/gh/HarshCasper/gidgethub/).

There are plenty of other issues in relation to Bug Discovery and Anti-Patterns which you would be interested to take a look at.

If you do not want to use DeepSource to continuously analyze this repo, I'll remove the `.deepsource.toml` from this PR and you can merge the rest of the fixes. If you want to setup DeepSource for Continuous Analysis, I can help you set that up.

### CODE REVIEW

File: gidgethub/__init__.py [hunk 1]
```diff
@@ -114,7 +114,8 @@ class BadGraphQLRequest(GraphQLException):
     """A 4XX HTTP response."""
 
     def __init__(self, status_code: http.HTTPStatus, response: Any) -> None:
-        assert 399 < status_code < 500
+        if 399 >= status_code:
```
Comment (Mariatta) [inline gidgethub/__init__.py:117]: This is a change in behavior, not a refactoring.

File: tests/test_sansio.py [hunk 2]
```diff
@@ -180,7 +180,7 @@ def test_all_keys_lowercase(self):
             user_agent, accept=test_api, oauth_token=oauth_token
         )
         assert len(headers) == 3
-        for key in headers.keys():
+        for key in headers:
```
Comment (Mariatta) [inline tests/test_sansio.py:183]: This also appears to be a change in behavior.

**Review by Mariatta — CHANGES_REQUESTED** [review]
I'm not sure about adopting DeepSource.

The fixes here broke tests and dropped code coverage, so it doesn't seem like it's improving anything, just extra maintenance burden.

At the moment I'm inclined not to accept the change.

**Comment by brettcannon** [comment]
I agree with @Mariatta that I am not seeing improvements from this. And those default values are on purpose to simplify the code; they are not dangerous as none of them are actually returned to allow someone to mutate their value.

**Comment by brettcannon** [comment]
@HarshCasper thanks for at least trying!

**Comment by srijan-deepsource** [comment]
Hey @brettcannon and @Mariatta! 
I just came across this PR and noticed that DeepSource generated a wrong fix. This particular bug was taken care of in our past releases.

I apologize for wasting your time on this.

**Pull request closed without merging**